src/main/java/org/jabref/gui/edit/automaticfiededitor/copyormovecontent/CopyOrMoveFieldContentTabView.java

```diff
@@ -52,23 +53,23 @@ public TwoFieldsTabView(List<BibEntry> selectedEntries, BibDatabaseContext datab
     }
 
     public void initialize() {
-        viewModel = new TwoFieldsViewModel(selectedEntries, databaseContext, dialogEdits);
+        viewModel = new CopyOrMoveFieldContentTabViewModel(selectedEntries, databaseContext.getDatabase(), dialogEdits);
         initializeFromAndToComboBox();
 
-        viewModel.overwriteNonEmptyFieldsProperty().bindBidirectional(overwriteNonEmptyFields.selectedProperty());
+        viewModel.overwriteFieldContentProperty().bindBidirectional(overwriteFieldContentCheckBox.selectedProperty());
 
-        moveValueButton.disableProperty().bind(viewModel.overwriteNonEmptyFieldsProperty().not());
-        swapValuesButton.disableProperty().bind(viewModel.overwriteNonEmptyFieldsProperty().not());
+        moveContentButton.disableProperty().bind(viewModel.overwriteFieldContentProperty().not());
+        swapContentButton.disableProperty().bind(viewModel.overwriteFieldContentProperty().not());
     }
 
     private void initializeFromAndToComboBox() {
-        fromFieldComboBox.getItems().addAll(viewModel.getAllFields().sorted(Comparator.comparing(Field::getName)));
-        toFieldComboBox.getItems().addAll(viewModel.getAllFields().sorted(Comparator.comparing(Field::getName)));
+        populateFieldsComboBox();
+        viewModel.getAllFields().addListener((InvalidationListener) observable -> populateFieldsComboBox());
```

Comment:
Why not directly make this an observableList? in the viewModel:
See 
https://github.com/JabRef/jabref/blob/d3e0f86a6be977aa16204fd611023e2358a556a7/src/main/java/org/jabref/gui/customentrytypes/CustomEntryTypeDialogViewModel.java#L52

Then you create a property for the ObserableList in the viewModel liike:
```
   public ObservableList<Field> fieldsForAdding() {
        return this.fieldsForAdding;
    }
```
And here in the View:
`        fromFieldComboBox.setItems(viewModel.fieldsForAdding());`

Reply:
Will `setItems()` listen for changes in `viewModel.fieldsForAdding()`?

Reply:
The question is: Do you need that? The field list should be more or less constant over the time. The user can always close and reopen the dialog if he changed some fields or added some
setItems anyway accepts an ObservableList

Reply:
I believe at least new fields should be added to the combo box. Removing unused fields is unnecessary and a potential source of bugs; the exception below is probably caused by it.  I'm just imaging myself setting a value for a new field and then I try to rename it, but the combo box doesn't show it. It doesn't sound right to me

Reply:
Ugh.. Even only adding new fields triggered some bugs. I'm removing the binding and following your suggestion. Additionally, I'll make all combo boxes editable so that users can simply type any missing fields.